File: apps/backend/api/src/presentation/mobile-dtos.ts
import { z } from 'zod';

export const MobileToken = z.object({ id: z.string(), symbol: z.string(), name: z.string() });
export type MobileTokenT = z.infer<typeof MobileToken>;

export const MobileAccount = z.object({
  id: z.string(),
  name: z.string(),
  typeId: z.string(),
  institutionId: z.string().nullable(),
  totalValue: z.string(),
});

export const MobileHolding = z.object({
  id: z.string(),
  accountId: z.string(),
  symbol: z.string(),
  name: z.string(),
  amount: z.string(),
  value: z.string().nullable(),
});

export const MobileGroup = z.object({
  id: z.string(),
  name: z.string(),
  color: z.string(),
  description: z.string().nullable(),
});

export type MobileGroupT = z.infer<typeof MobileGroup>;

export const MobileVault = z.object({
  id: z.string(),
  name: z.string(),
  targetAmount: z.string(),
  currentAmount: z.string(),
  currencyId: z.string(),
  color: z.string(),
  iconName: z.string().nullable(),
  description: z.string().nullable(),
});

export type MobileVaultT = z.infer<typeof MobileVault>;

File: apps/mobile/shared/src/commonMain/kotlin/xyz/scani/mobile/shared/data/MobileDtos.kt
package xyz.scani.mobile.shared.data

import kotlinx.serialization.Serializable

@Serializable
data class MobileAccount(
    val id: String,
    val name: String,
    val typeId: String,
    val institutionId: String? = null,
    val totalValue: String,
)

@Serializable
data class MobileHolding(
    val id: String,
    val accountId: String,
    val symbol: String,
    val name: String,
    val amount: String,
    val value: String? = null,
)

@Serializable
data class MobileGroup(
    val id: String,
    val name: String,
    val color: String,
    val description: String? = null,
)

@Serializable
data class MobileVault(
    val id: String,
    val name: String,
    val targetAmount: String,
    val currentAmount: String,
    val currencyId: String,
    val color: String,
    val iconName: String? = null,
    val description: String? = null,
)

@Serializable
data class MobileToken(val id: String, val symbol: String, val name: String)
